packages/block-editor/src/components/link-control/index.js

```diff
@@ -176,77 +159,66 @@ function LinkControl( {
+						{ __( 'Currently selected' ) }:
+					</p>
+					<div
+						aria-labelledby={ `current-link-label-${ instanceId }` }
+						aria-selected="true"
+						className={ classnames( 'block-editor-link-control__search-item', {
+							'is-current': true,
+						} ) }
+					>
+						<span className="block-editor-link-control__search-item-header">
+							<ExternalLink
+								className="block-editor-link-control__search-item-title"
+								href={ value.url }
 							>
-								<span className="block-editor-link-control__search-item-header">
-									<ExternalLink
-										className="block-editor-link-control__search-item-title"
-										href={ value.url }
-									>
-										{ value.title }
-									</ExternalLink>
-									<span className="block-editor-link-control__search-item-info">{ filterURLForDisplay( safeDecodeURI( value.url ) ) || '' }</span>
-								</span>
-
-								<Button isSecondary onClick={ setMode( MODE_EDIT ) } className="block-editor-link-control__search-item-action block-editor-link-control__search-item-action--edit">
-									{ __( 'Edit' ) }
-								</Button>
-							</div>
-						</Fragment>
-					) }
-
-					{ isEditingLink && (
-						<LinkControlSearchInput
-							value={ inputValue }
-							onChange={ onInputChange }
-							onSelect={ ( suggestion ) => {
-								setIsEditingLink( false );
-								onChange( { ...value, ...suggestion } );
-							} }
-							renderSuggestions={ renderSearchResults }
-							fetchSuggestions={ getSearchHandler }
-							onReset={ resetInput }
-						/>
-					) }
-
-					{ ! isEditingLink && (
-						<LinkControlSettingsDrawer value={ value } settings={ settings } onChange={ onChange } />
-					) }
-				</div>
-			</div>
-		</Popover>
+								{ value.title }
+							</ExternalLink>
+							<span className="block-editor-link-control__search-item-info">{ filterURLForDisplay( safeDecodeURI( value.url ) ) || '' }</span>
+						</span>
+
+						<Button isSecondary onClick={ setMode( MODE_EDIT ) } className="block-editor-link-control__search-item-action block-editor-link-control__search-item-action--edit">
+							{ __( 'Edit' ) }
+						</Button>
+					</div>
+				</Fragment>
+			) }
+
+			{ isEditingLink && (
+				<LinkControlSearchInput
+					value={ inputValue }
+					onChange={ onInputChange }
+					onSelect={ ( suggestion ) => {
+						setIsEditingLink( false );
+						onChange( { ...value, ...suggestion } );
+					} }
+					renderSuggestions={ renderSearchResults }
+					fetchSuggestions={ getSearchHandler }
+					onReset={ resetInput }
+				/>
+			) }
+
+			{ ! isEditingLink && (
+				<LinkControlSettingsDrawer value={ value } settings={ settings } onChange={ onChange } />
+			) }
+		</div>
 	);
 }
 
-export default compose(
-	withInstanceId,
-	withSelect( ( select, ownProps ) => {
-		if ( ownProps.fetchSearchSuggestions && isFunction( ownProps.fetchSearchSuggestions ) ) {
-			return;
-		}
-
+function ConnectedLinkControl( props ) {
```

Comment:
I don't like that I had to do that to make unit tests pass (test the unconnected component). Ideally, we'd find a way to easily mock useSelect

Reply:
> I don't like that I had to do that to make unit tests pass (test the unconnected component). Ideally, we'd find a way to easily mock useSelect

Follow-up at #19705